Client/src/view/BattleView/MapBox.java

```diff
@@ -258,59 +261,92 @@ void updateMapColors() {
-                        cells[row][column].setFill(Constants.CAN_SELECT_COLOR);
-                    } else
-                        cells[row][column].setFill(Constants.defaultColor);
-                    continue;
-                }
-                if (selectedTroop != null && selectedTroop.getCell().getRow() == row &&
-                        selectedTroop.getCell().getColumn() == column) {
-                    cells[row][column].setFill(Constants.SELECTED_COLOR);//not important
-                    continue;
+                if (selectionType == SelectionType.NORMAL){
+                    boolean updateNormal = updateMapColoursOnNormal(row, column, player);
+                    if (updateNormal) { continue;}
                 }
 
-                if (selectionType == SelectionType.SPELL) {
-                    cells[row][column].setFill(Constants.defaultColor);
+                if (selectedTroop != null && selectedTroop.getCell() == new Cell(row, column)){
+                    cells[row][column].setFill(Constants.SELECTED_COLOR);
                     continue;
                 }
-                if (selectionType == SelectionType.NORMAL) {
-                    if (GameController.getInstance().getAvailableActions().canAttack(gameMap, player, selectedTroop, row, column))
-                        cells[row][column].setFill(Constants.ATTACK_COLOR);
-                    else if (GameController.getInstance().getAvailableActions().canMove(gameMap, player,
-                            selectedTroop, row, column))
-                        cells[row][column].setFill(Constants.MOVE_COLOR);
-                    else
-                        cells[row][column].setFill(Constants.defaultColor);
+
+                updateMapColourHighlightEnemyUnits(row, column, player);
+                updateMapColoursHighlightUnitActions(row, column, player);
+            }
+        }
+    }
+
+    private void updateMapColoursHighlightUnitActions(int row, int column, CompressedPlayer player){
+        Troop troop = getTroop(row, column);
+        if (troop == null){ return; }
+        if (!battleScene.isMyTurn()) { return; }
+        if (troop.getPlayerNumber() != player.getPlayerNumber()) { return;}
+
+        if (troop.canMove()){
+            cells[row][column].setFill(Constants.CAN_MOVE);
+        }
+        //if (troop.canAttack()){
+        //    cells[row][column].setFill(Constants.CAN_ATTACK);
+        //}
+    }
+
+    private void updateMapColourHighlightEnemyUnits(int row, int column, CompressedPlayer player){
+        Troop troop = getTroop(row, column);
+        if (troop == null){ return; }
+
+        if (battleScene.isMyTurn()){
+            if (troop.getPlayerNumber() != player.getPlayerNumber()){
+                cells[row][column].setFill(Constants.ENEMY_UNIT);
+            }
+        }
+        else{
+            if (troop.getPlayerNumber() == player.getPlayerNumber())
+                cells[row][column].setFill(Constants.ENEMY_UNIT);
+        }
+    }
```

Comment:
```suggestion
        if (troop.getPlayerNumber() != this.battleScene.getMyPlayerNumber()) {
            cells[row][column].setFill(Constants.ENEMY_UNIT);
        }
```

This compiled for me but I haven't been able to get the client to run on this computer to test it.

Reply:
"haven't been able to get the client to run"

how do you mean. can't as in client crashed or something?  or cant as in lack of time, or something?

Reply:
As in you should try this change locally and see if it's the same for you.

On this computer the client built and started up, but when I try to login (either to central or localhost) nothing happens. No errors, but nothing else changes either. I don't have time today to diagnose the issue, so I made the suggestion anyway since it compiles/builds and starts.